### ISSUE TITLE
CVC4 Crash (core dumped)

### ISSUE THREAD

**Issue by farif** [CVC4/CVC4#5809]
**Describe the bug**
A bug (core-dump crash) is triggered and crash CVC4 when a simple specification is provided that use AUFLIA logic.    
The bug is triggered with develop version.

**Command line arguments**: --lang=sygus2
**CVC4 version/commit**: This is CVC4 version 1.9-prerelease [git master e0dfc0a3]
**Operating system**: Ubuntu 20.04.1 LTS

The specification file is attached that triggers the bug:
fatal failure within CVC4::api::Sort CVC4::SymbolTable::Implementation::lookupType(const string&, const std::vector<CVC4::api::Sort>&) const at /home/farif/Documents/projects/CVC4/src/expr/symbol_table.cpp:535
Unhandled case encounteredCould not instantiate sort
Aborted (core dumped)
farif@farif-machine:~_

**Comment by farif** [comment]
```
lang=sygus2
(set-logic ALIA)
(set-option :sygus-out status-and-def)
(set-option :e-matching false)
(define-sort Stream (T) (Array Int T))
(define-sort IntStream () (Stream Int))
(synth-fun phi ((t IntStream) ) IntStream
   ((<F> IntStream))
   ((<F> IntStream (
       ((as const IntStream) 0)
    ))
   )
)
(check-synth)

```

**Comment by daniel-larraz** [comment]
FYI, it works fine with CVC4 1.8.

**Comment by farif** [comment]
The above issue is reported for the following version:
This is CVC4 version 1.9-prerelease [git master e0dfc0a3]
compiled with GCC version 9.3.0
on Jan  4 2021 18:09:40

**Comment by HanielB** [comment]
Fixed in #5810
